client/lib/pages/plant-list.dart

```diff
@@ -127,9 +132,70 @@ class _PlantListState extends State<PlantList> {
                   child:
                   (){
                     if(plantListItems != null){
-                      return ListView(
-                        children: plantListItems,
+                      return Column(
+                        children: <Widget>[
+                          Expanded(
+                            flex: MediaQuery.of(context).orientation == Orientation.portrait ? 1 : 1,
+                            child: Padding(
+                              padding: EdgeInsets.all(16),
+                              child: TextField(
+                                controller: _filterInputController,
+                                decoration: InputDecoration(
+                                  focusedBorder: UnderlineInputBorder(
+                                    borderSide: BorderSide(color: theme.accentColor),
+                                  ),
+                                  enabledBorder: UnderlineInputBorder(
+                                      borderSide: BorderSide(color: theme.accentColor)
+                                  ),
+                                  hintText: 'Zoek op naam',
+                                  suffixIcon: IconButton(
+                                    icon: Icon(Icons.clear),
+                                    onPressed: () {
+                                      _filterInputController.clear();
+                                      filteredPlantListItems = null;
+                                      setState(() {});
+                                    },
+                                    color: Colors.black,
+                                  )
+                                ),
+                                onChanged: (String filter) {
+                                  filter = filter.trimRight().toLowerCase();
```

Comment:
You trim on the right. But if you start a string with whitespaces it will not ignore it.